### PR TITLE
WIP: Add JSDoc to utils directory scripts

### PR DESCRIPTION
**Pull request by dcgoodwin2112**
## Description
Work in progress. Add JSDoc to to utility functions.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#31481

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/utils/api/index.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import { createApiEvent } from '../analytics';
 import recordEvent from 'platform/monitoring/record-event';
 
+/**
+ * @param {Promise} request
+ * @param {string} [eventName]
+ * @param {string} [token]
+ */
```
Comment (acrollet) [inline src/applications/check-in/utils/api/index.js:8]: Are you only specifying the return type where the type checker can't infer it? If so maybe we should also specify that in the readme so we can do things consistently 🙂

**Review by acrollet — CHANGES_REQUESTED** [review]
This is awesome! The DX improvement is great. I'd love to see a section added to the README so that new devs can know how to get set up for this.

File: src/applications/check-in/utils/api/index.js (hunk 1 above)
Comment (dcgoodwin2112) [reply src/applications/check-in/utils/api/index.js:8]: Thanks @acrollet! You are absolutely correct about the return types and I'll update readme. The type-checker in the editor is really good at inferring types for code that is self-contained, so I only included types where types could not be identified without them.

Comment (acrollet) [reply src/applications/check-in/utils/api/index.js:8]: sweet!

**Review by acrollet — APPROVED** [review]

**Pull request merged**